test/crowdsale/Crowdsale.test.js

```diff
@@ -15,67 +16,118 @@ contract('Crowdsale', function ([_, investor, wallet, purchaser]) {
   const value = ether(42);
   const tokenSupply = new BigNumber('1e22');
   const expectedTokenAmount = rate.mul(value);
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
 
-  beforeEach(async function () {
-    this.token = await SimpleToken.new();
-    this.crowdsale = await Crowdsale.new(rate, wallet, this.token.address);
-    await this.token.transfer(this.crowdsale.address, tokenSupply);
+  it('requires a valid token', async function () {
```

Comment:
I'd say "non zero" instead of "valid". Validity is a much stronger property, and sometimes poorly defined. This applies to many of the tests added in this PR, please review them all.